command/connect/envoy/envoy.go

```diff
@@ -24,6 +24,12 @@ import (
 )
 
 func New(ui cli.Ui) *cmd {
+	if bui, ok := ui.(*cli.BasicUi); ok {
```

Comment:
I originally started going down this short list and I scrapped that and did it completely differently. I did not make an attempt to shrink the interface down mostly because that likely should happen uniformly across the whole CLI rather than in a one-off like this.

Reply:
Sounds good, I like the approach you took here